Set custom SecurityManager constructor accessible flag

JasonFengJ9 opened the pull request:
**Set custom SecurityManager constructor accessible flag**

The constructor for custom `SecurityManager` must set its accessible flag to `true` to suppress access checking, otherwise an `IllegalAccessException` is thrown when invoking `constructor.newInstance()`.
Changed `getConstructor()` to `getDeclaredConstructor()` to only allow instantiation of custom `SecurityManger` with a `public constructor` just like `RI`.
Added a test to ensure that `NoSuchMethodException` is thrown at `getConstructor()` when loading a custom `SecurityManger` with a `non-public constructor`.

Verified that this PR passes the test at https://github.com/eclipse/openj9/issues/6661

closes: #6661 

Reviewer: @pshipton 
FYI: @DanHeidinga 

Signed-off-by: Jason Feng <fengj@ca.ibm.com>

pshipton commented:
We can update TestSecurityManager in test/functional/Java8andUp/src_`*`/org/openj9/test/java/lang/Test_System.java to test this case.

JasonFengJ9 commented:
> test/functional/Java8andUp/src_*/org/openj9/test/java/lang/Test_System.java

The failure was due to access check against a custom `SecurityManager` class within a module which is `modularity` specific hence it doesn't apply to `Java 8`.
To reproduce this failure for `JDK 11+`, a customer `SecurityManager` module is required to be built and probably run via a `cmdline_tester`, which will look similar with this testcase.
Since this is a function test, I am leaning toward using the JTReg test instead.

pshipton commented:
I just meant we can add a non-public constructor to the existing test.

JasonFengJ9 commented:
Changed `getConstructor()` to `getDeclaredConstructor()` to only allow instantiation of custom `SecurityManger` with a `public constructor` just like `RI`.
Added a test to ensure that `NoSuchMethodException` is thrown at `getConstructor()` when loading a custom `SecurityManger` with a `non-public constructor`.

Note that newly added test is for changed behaviours to get custom `SecurityManager` constructor. For the failure reported at #6661, JTReg test in question seems sufficient.

Please have another look.

pshipton commented on test/functional/Java8andUp/src/org/openj9/test/support/Support_Exec.java:
```diff
@@ -57,37 +80,57 @@ public static String execJava(String[] args, String[] classpath,
 			}
 		}
+		if (checkExpectedErr && !expectedErrFound) {
+			Assert.fail("Not found expected error string: " + expectedErr);
```
It might be useful to show the output which did occur.

JasonFengJ9 replied:
Appended the actual error output. Please review.

pshipton commented:
jenkins test sanity plinux jdk8,jdk11
